components/business-adaptors/sap/src/main/java/org/wso2/carbon/transports/sap/SAPTransportSender.java

```diff
@@ -152,22 +152,44 @@ public void sendMessage(MessageContext messageContext, String targetEPR,
                 destination.confirmTID(tid);
 
             } else if (uri.getScheme().equals(SAPConstants.SAP_BAPI_PROTOCOL_NAME)) {
+                String escapeErrorHandling = (String) messageContext.getProperty(SAP_ESCAPE_ERROR_HANDLING);
                 try {
                     OMElement payLoad,body;
                     body = messageContext.getEnvelope().getBody();
                     payLoad = body.getFirstChildWithName(new QName(RFCConstants.BAPIRFC));
                     log.info("Received RFC/Meta DATA: " + payLoad);
                     String rfcFunctionName = RFCMetaDataParser.getBAPIRFCFucntionName(payLoad);
                     log.info("Looking up the BAPI/RFC function: " + rfcFunctionName + ". In the " +
-                             "meta data repository");
-
-                    JCoFunction function = getRFCfunction(destination, rfcFunctionName);
-                    RFCMetaDataParser.processMetaDataDocument(payLoad, function);
-                    String escapeErrorHandling = (String) messageContext.getProperty(SAP_ESCAPE_ERROR_HANDLING);
-                    String responseXML = evaluateRFCfunction(function, destination, escapeErrorHandling);
+                            "meta data repository");
+                    String responseXML;
+                    if (isTransaction(messageContext)) {
+                        JCoContext.begin(destination);
+                        log.info("Begin transaction.");
+                        JCoFunction function = getRFCfunction(destination, rfcFunctionName);
+                        RFCMetaDataParser.processMetaDataDocument(payLoad, function);
+                        responseXML = evaluateRFCfunction(function, destination, escapeErrorHandling);
+                        JCoFunction commitFunction = getRFCfunction(destination, SAPConstants.BAPI_TRANSACTION_COMMIT);
+                        evaluateRFCfunction(commitFunction, destination, escapeErrorHandling);
+                        log.info("Commit transaction.");
+                    } else {
+                        JCoFunction function = getRFCfunction(destination, rfcFunctionName);
+                        RFCMetaDataParser.processMetaDataDocument(payLoad, function);
+                        responseXML = evaluateRFCfunction(function, destination, escapeErrorHandling);
```

Comment:
These 3 lines are common to both if and else conditions. Can we move it to a separate method?

Reply:
We can move. But I am afraid about the code readability. I feel method like below not make elegant code because `evaluateRFCfunction` method does the same action.

`private String getResponse(JCoDestination destination, String escapeErrorHandling, OMElement payLoad, String rfcFunctionName) throws AxisFault `